pkg/provision/common/doc.go

```diff
@@ -0,0 +1,3 @@
+// Package common hold logic that is used by both the provision and primitive package
+// it purpose is mainly to avoid circular dependencies
+package common
```

Comment:
seeing the evolution of this PR it feels to me like this package would no longer be needed?

Reply:
It still is, without it we still circular dependency between the primitives and provision package.